tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source.Tests/DiagnosticAnalysisRunnerTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac.Extras.Moq;
+using AutoFixture;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.MSBuild;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Source.Tests
+{
+    public class DiagnosticAnalysisRunnerTests
+    {
+        private readonly Fixture _fixture;
+        internal const string TestProjectPath = @"assets\TestProject.csproj";
+
+        private static IEnumerable<AdditionalText> AdditionalTexts
+        {
+            get
+            {
+                return Array.Empty<AdditionalText>();
+            }
+        }
+
+        public DiagnosticAnalysisRunnerTests()
+        {
+            _fixture = new Fixture();
+        }
+
+        [Fact]
+        public async Task DiagnosticsWithEmptyProjectTest()
+        {
+            // Arrange
+            using var mock = AutoMock.GetLoose();
+            var loggerMock = new Mock<ILogger<DiagnosticAnalysisRunner>>();
+
+            var project = CreateProject(mock);
+            project.Setup(p => p.GetRoslynProject()).Returns(() =>
+            {
+                var ws = new AdhocWorkspace();
+                var name = Path.GetFileNameWithoutExtension(TestProjectPath)!;
+                return ws.AddProject(ProjectInfo.Create(ProjectId.CreateNewId(), VersionStamp.Default, name, name, "C#", filePath: TestProjectPath));
+            });
+
+            var context = mock.Mock<IUpgradeContext>();
+            context.Setup(c => c.Projects).Returns(new[] { project.Object });
+            var diagnosticAnalysisRunner = new DiagnosticAnalysisRunner(GetAnalyzers(false), AdditionalTexts, loggerMock.Object);
```

Comment:
You shouldn't have to create it like this. We use Automock so you can just do `mock.Create<DiagnosticAnalysisRunner>()` and it will compose it correctly

Reply:
I tried that first, but it was not able to pass in the parameters for `DiagnosticAnalysisRunner` - Analyzers. AdditionalTexts, etc.